server/apps/archive/archive_crop.py

```diff
@@ -124,22 +127,30 @@ def _add_crop(self, original, crop_name, doc):
         """
         renditions = original.get('renditions')
         crop_data = UploadService().get_cropping_data(doc)
-        original_image = original.get('renditions').get('original')
-        original_file = superdesk.app.media.get(original_image.get('media'), 'upload')
-        if not original_file:
-            raise SuperdeskApiError.badRequestError('Original file couldn\'t be found')
+        original_crop = renditions.get(crop_name, {})
+        if (crop_data[0] != original_crop.get('CropLeft') or crop_data[1] != original_crop.get('CropTop') or
+                crop_data[2] != original_crop.get('CropRight') or crop_data[3] != original_crop.get('CropBottom')):
 
-        try:
-            cropped, out = crop_image(original_file, original_file.filename, crop_data)
-            if cropped:
-                renditions[crop_name] = self._save_cropped_image(out, original_file, doc)
-                get_resource_service(ARCHIVE).patch(original['_id'], {'renditions': renditions})
-            else:
-                raise SuperdeskApiError.badRequestError('Saving crop failed: {}'.format(str(out)))
-        except SuperdeskApiError:
-            raise
-        except Exception as ex:
-            raise SuperdeskApiError.badRequestError('Generating crop failed: {}'.format(str(ex)))
+            original_image = original.get('renditions').get('original')
+            original_file = superdesk.app.media.get(original_image.get('media'), 'upload')
+            if not original_file:
+                raise SuperdeskApiError.badRequestError('Original file couldn\'t be found')
+
+            try:
+                cropped, out = crop_image(original_file, original_file.filename, crop_data)
+                if cropped:
+                    renditions[crop_name] = self._save_cropped_image(out, original_file, doc)
+                    if original.get(ITEM_STATE) in PUBLISH_STATES:
+                        get_resource_service(ARCHIVE).system_update(original[config.ID_FIELD],
+                                                                    {'renditions': renditions}, original)
+                    else:
+                        get_resource_service(ARCHIVE).patch(original[config.ID_FIELD], {'renditions': renditions})
+                else:
+                    raise SuperdeskApiError.badRequestError('Saving crop failed: {}'.format(str(out)))
```

Comment:
IMO it would be a bit more readable if we reverse the condition and move the shorter branch up - the main block would not have to be indented for an extra level anymore and the code would read more linearly and with less nesting.

```py
...
if not cropped:
    raise SuperdeskApiError.badRequestError(...)

# the main block, now outdented
renditions[crop_name] = self._save_cropped_image(...)
...
```